tests/UnifiedSpecTests/Operation.php

```diff
@@ -368,18 +435,31 @@ private function executeForCollection(Collection $collection)
                 $options = isset($args['options']) ? (array) $args['options'] : [];
                 $options += array_diff_key($args, ['documents' => 1]);
 
+                assertArrayHasKey('documents', $args);
+                assertIsArray($args['documents']);
+
                 return $collection->insertMany(
                     $args['documents'],
                     $options
                 );
             case 'insertOne':
+                assertArrayHasKey('document', $args);
+                assertInstanceOf(stdClass::class, $args['document']);
+
                 return $collection->insertOne(
                     $args['document'],
                     array_diff_key($args, ['document' => 1])
                 );
             case 'listIndexes':
-                return $collection->listIndexes($args);
+                return iterator_to_array($collection->listIndexes($args));
             case 'mapReduce':
+                assertArrayHasKey('map', $args);
+                assertArrayHasKey('reduce', $args);
+                assertArrayHasKey('out', $args);
+                assertInstanceOf(Javascript::class, $args['map']);
+                assertInstanceOf(Javascript::class, $args['reduce']);
```

Comment:
We could technically use JavascriptInterface here (per MapReduce's own signature), but decoding extended JSON should always yield Javascript objects. I don't believe any existing unified tests cover mapReduce, but the tests in retryable reads do use `$code` for this type -- so I'd expect these to still be Javascript whenever those are ported to the unified format.